### PR TITLE
[spirv] support half zero constant correctly

### PR DESCRIPTION
**Pull request by jaebaek**

### CODE REVIEW

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 1]
```diff
@@ -9097,17 +9097,15 @@ SpirvConstant *SpirvEmitter::getValueZero(QualType type) {
...
     if (isScalarType(type, &scalarType)) {
       if (scalarType->isSignedIntegerType()) {
-        return spvBuilder.getConstantInt(astContext.IntTy, llvm::APInt(32, 0));
+        return spvBuilder.getConstantInt(scalarType, llvm::APInt(32, 0));
       }
 
       if (scalarType->isUnsignedIntegerType()) {
-        return spvBuilder.getConstantInt(astContext.UnsignedIntTy,
-                                         llvm::APInt(32, 0));
+        return spvBuilder.getConstantInt(scalarType, llvm::APInt(32, 0));
```
Comment (antiagainst) [inline tools/clang/lib/SPIRV/SpirvEmitter.cpp]: This case can be merged into the previous case now?

Comment (jaebaek) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: Aha right. Nice catch :)

Comment (jaebaek) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: Updated

**Comment by jaebaek** [comment]
Fixes #2044

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp (hunk 1 above)
Comment (ehsannas) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: you can use
`scalarType->isIntegerType()` which checks for both signed and unsigned integers.
However, both `isUnsignedIntegerType()` and `isIntegetType()` return true for Booleans.

So I think it'd be best to use
`if(scalarType->isIntegerType() && !scalarType->isBooleanType())`

Comment (jaebaek) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: Now it handles boolean case in a separate branch. PTAL.

**Pull request merged**